device.go

```diff
@@ -197,6 +197,11 @@ func (device *GenericDevice) deviceType() string {
 
 // isVFIO checks if the device provided is a vfio group.
 func isVFIO(hostPath string) bool {
+	// Ignore /dev/vfio/vfio character device
+	if strings.HasPrefix(hostPath, filepath.Join(vfioPath, "vfio")) {
```

Comment:
This works but I thought that we were not trying to handle privileged containers with virtualized containers since the whole security is compromised with such containers.

Reply:
We can't truly run in privileged mode, but when starting a container with --privileged, Docker does pass in a bunch of devices that may be required to run as a privileged container. We need to handle them.

Reply:
But don't you think this should maybe handled at the runtime level and not in virtcontainers ?

Reply:
What do you mean? We handle the devices in virtcontainers, we were wrongly classifying "/dev/vfio/vfio" as a device bound  to the vfio driver when in fact it is a special device created by the kernel module. This commit rectifies this bug.

Reply:
No I understand this is working and what it does, but I was thinking that if we don't expect to see this device/path at all since this is not a valid case, this should not be passed by the runtime as a device. I was only suggesting that our runtime would not allow such a device path to be provided to virtcontainers, but I might be wrong here.

Reply:
The runtime just passes the configuration to virtcontainers, it is upto the library how to handle the device. I don't think the runtime should handle what devices need to be passed to the library.

Reply:
Fair enough

Reply:
my 2p worth - I agree with the idea that as we cannot fully support `--privileged` mode (from the runtime and/or in virtcontainers either), that I'd much rather the runtime gracefully fails with a clear message if invoked with/in `--privileged` mode. Otherwise we are just misleading the end user who has run with `--privileged`, sees their container come up, but may be tracking down 'why did this not work as expected' :-(
I think I'm agreeing with you both:
- I'm happy for the virtcontainers to drop paths it knows it cannot support (I wonder if we should issue a log warning or message along with that if we don't already)
- I'd really like the runtime to gracefully fail to run in `--privileged` mode - but, I don't know if the runtime can even tell from the json if it is in that mode?

I'd like to hear if @sameo or @mcastelino have any thoughts on this (the runtime gracefully failing). Let me go open a runtime ticket (and tie it to the runtime limitations page) - as I think this is a runtime thing, not really a virtcontainers thing.

Reply:
For reference: https://github.com/clearcontainers/runtime/issues/746

Reply:
@grahamwhaley @amshinde we should log failure to map specific devices we do not support. But we always ignore the standard set the comes for all containers. To log them would make the log less useful. But we should log anything outside the standard set that we do not suppport.

Reply:
@mcastelino there is no way of knowing programatically what devices are passed on the command line whereas those passed by docker itself, unless we maintain a a list of them of what we think are standard ones.